### ISSUE TITLE
Bug: yarn install --production=false will install the dependencies

### ISSUE THREAD

**Issue by SherrybabyOne** [yarnpkg/yarn#8829]
yarn@1.22.18

As an example, package.json:

```json
{
  "name": "node-test",
  "version": "1.0.0",
  "dependencies": {
    "color": "4.2.3"
  },
  "devDependencies": {
    "prettier": "2.6.2"
  }
}
```

Execute `yarn install --production=false` on the command line, and both `colors` and `prettier` will be installed, and it has been checked that prettier does not require colors dependency

**Comment by markcellus** [comment]
This seems like expected behavior. [Per docs](https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-production-true-false), setting `production` to `true` should only install packages in `dependencies` and setting it to `false` would also install  packages in`devDependencies`. What were you expecting?

**Comment by bloudermilk** [comment]
I agree with the OP that there is no clear alternative to overriding the `NODE_ENV` value now that `--production=false` doesn't exist.
